core/src/databases/database.rs

```diff
@@ -83,20 +84,227 @@ impl Database {
+                        rows.into_par_iter()
+                            .map(|(table, r)| {
+                                Ok((
+                                    table.table_id().to_string(),
+                                    DatabaseSchemaTable::new(table, TableSchema::from_rows(&r)?),
+                                ))
+                            })
+                            .collect::<Result<HashMap<_, _>>>()?,
+                    ),
+                    returned_rows,
                 ))
             }
         }
     }
 
+    pub async fn create_in_memory_sqlite_conn(
+        &self,
+        project: &Project,
+        store: Box<dyn Store + Sync + Send>,
+    ) -> Result<Connection> {
+        match self.db_type {
+            DatabaseType::REMOTE => Err(anyhow!(
+                "Cannot build an in-memory SQLite DB for a remote database."
+            )),
+            DatabaseType::LOCAL => {
+                let time_build_db_start = utils::now();
+                let (schema, rows_by_table) = self.get_schema(project, store.clone(), true).await?;
+                let rows_by_table = match rows_by_table {
+                    Some(rows) => rows,
+                    None => return Err(anyhow!("No rows found")),
+                };
+                utils::done(&format!(
+                    "DSSTRUCTSTAT Finished retrieving schema: duration={}ms",
+                    utils::now() - time_build_db_start
+                ));
+
+                let table_schemas: HashMap<String, TableSchema> = schema
+                    .iter()
+                    .filter(|(_, table)| !table.schema.is_empty())
+                    .map(|(table_name, table)| (table_name.clone(), table.schema.clone()))
+                    .collect();
+
+                let generate_create_table_sql_start = utils::now();
+                let create_tables_sql: String = schema
+                    .iter()
+                    .filter(|(_, table)| !table.schema.is_empty())
+                    .map(|(table_name, table)| {
+                        table
+                            .schema
+                            .get_create_table_sql_string(table_name.as_str())
+                    })
+                    .collect::<Vec<_>>()
+                    .join("\n");
+                utils::done(&format!(
+                    "DSSTRUCTSTAT Finished generating create table SQL: duration={}ms",
+                    utils::now() - generate_create_table_sql_start
+                ));
+
+                let conn = rusqlite::Connection::open_in_memory()?;
+
+                let create_tables_execute_start = utils::now();
+                conn.execute_batch(&create_tables_sql)?;
+                utils::done(&format!(
+                    "DSSTRUCTSTAT Finished creating tables: duration={}ms",
+                    utils::now() - create_tables_execute_start
+                ));
+
+                let insert_execute_start = utils::now();
+                rows_by_table
```

Comment:
as this discussed this should be parallelized all the way to creating the arguments of conn.execute

Reply:
It's easy to // when dealing with string but with `ToSql` params it's hard because it cannot be send through threads. I was thinking of making my own enum that can be easily converted into ToSQl

Reply:
I will try that and call for help if I fail

Reply:
The experiment code managed to do so no?

Reply:
No it was `format`ing strings, which isn't OK here for audit reasons

Reply:
made it work :)

Reply:
So now what I do:
- iterate over each table sequentially (we can't `prepare` statement in // and we cant execute in //)
- for each table, create the params vec for rows in //
- then for each created params vec, insert rows sequentially. 

Do you think I can do even better ?